backend/controllers/submissionController.js

```diff
@@ -39,6 +40,73 @@ const getTaskSubmissionsByUser = async (req, res) => {
   res.status(200).send(submissions);
 };
 
+const getCompetitionLeaderboard = async (req, res) => {
+  const { name } = req.params;
+
+  const competition = await CompetitionModel.findOne({ name });
+
+  if (!competition) {
+    res.status(404).send();
+    return;
+  }
+
+  const submissions = await SubmissionModel.aggregate([
+    {
+      $match: {
+        $expr: {
+          $and: [
+            {
+              $in: [
+                '$parentTask',
+                competition.tasks.map((task) => new ObjectId(task)),
+              ],
+            },
+            { $gt: ['$score', 0] },
+          ],
+        },
+      },
+    },
+    {
+      $group: {
+        _id: '$user',
+        createdAt: { $first: '$createdAt' },
+        score: { $sum: '$score' },
+      },
+    },
+    {
+      $match: {
```

Comment:
Gjør vi samme sjekken to ganger? Linje 64 og 78

Reply:
lol yeah. Fikset nå